File: 14-longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
    if len(strs) == 0{
        return ""
    }
    base := strs[0]
    idx := 0
    for ; idx < len(base); idx += 1{
        for _, str := range strs {
            if len(str) > idx && base[idx] == str[idx]{
                continue
            } else {
                return base[:idx]
            }
        }
    }
    return base[:idx]
}

File: bwu/string/14-longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        n = len(strs)
        res = ''
        if n == 0:
            return res
            
        minLen = len(strs[0])
        for s in strs:
            minLen = min(minLen, len(s))

        for i in xrange(minLen):
            ch = strs[0][i]
            for j in xrange(n):
                if strs[j][i] != ch:
                    return res
            res += ch
        return res
